### ISSUE TITLE
2019-2020 Hacknights brainstorming

### ISSUE THREAD

**Issue by ngiangre** [nyhais/hacknights#1]
@m3ngineer 

I figure we'll use the existing repo here, and have the issue tracker be our discussion platform. 

We'll discuss the content, but I think we can still organize under hacknights.

So here's what I propose we should move forward (but please object or anything I'm just suggesting):

1. Pick a methodological topic
2. Identify application areas of the above methodology
3. Organize recurring hacknights (TBD)

Do you agree with the above and having any modifications? Have you been able to talk to your contact?

Cheers,
Nick

**Comment by ngiangre** [comment]
@m3ngineer can you give me permission to push to the repo? I get
```
Nicholass-MacBook-Pro-3:hacknights npg2108$ git push
remote: Permission to nyhais/hacknights.git denied to ngiangre.
```

I'm adding a requirements.txt and making sure that, atleast within a clean virtualenv, if you install everything in requirements.txt with pip3 install ... you should be able to run the malaria notebook.

Then I want to start making notebooks to catalog our brainstorming. Sounds good?

**Comment by m3ngineer** [comment]
@ngiangre You're now added as a collaborator. Let me know if you have any issues.

**Comment by m3ngineer** [comment]
@ngiangre Check out this medium article on most used algorithms in healthcare literature.
https://medium.com/sciforce/top-ai-algorithms-for-healthcare-aa5007ffa330

I personally tend to see NNs and NLP used in the healthcare tech space because of the image and EHR data they are working with. I haven't seen much with SVMs, but maybe it's different in the literature. I'm thinking Bayesian classifiers, NNs or NLP would be pretty relevant topics.

**Comment by ngiangre** [comment]
Great, thanks for the article!

I agree with you. For image data, definitely NNs like CNNs are used (if researchers can get their hands on that data). For unstructured (text) data, a lot of named entity recognition was (and is) used but now RNNs that incorporate sentence and word structure are desirable. Since there are a lot of small patient numbers for those with particular diseases or in particular studies, bayesian methods would seem to help out in prediction tasks. 

My hope is that we can eventually cover a bit of genomic data in both it's meaning, data representation, and relevance to analyze and integrate with clinical data. I made a notebook called bioconductor because that organization (R language foundation) is the most open-source, centralized, and accessible place for genomic software and data. But that's just one source that we can highlight. 

I propose we try to scrape/compile/list open datasets we can use with these methods. We might be able to just grab stuff from kaggle, but I'm sure there's other sources. How does that sound? We can work on this for a few weeks and then reconvene.

**Comment by m3ngineer** [comment]
Sounds good. Let's check in August 3rd or 4th.

**Comment by ngiangre** [comment]
Hey @m3ngineer 

Check out the new urls and some processed datasets at Clinical_and_Genomic_Data_Sources.ipynb

Let me know what your updates are!

**Comment by m3ngineer** [comment]
@ngiangre I added a couple of datasets as well!

It seems like there's an abundance of open source image datasets available, while EHR/NLP-type datasets are more restrictive (except ct.gov or pubmed). I am cool doing something genomically-related (like maybe with SEER?), but I'm pretty unfamiliar with the techniques used, so I'll be depending on you to provide some initial guidance on selecting on technique for this.

**Comment by ngiangre** [comment]
Great!

And thanks for the slides! I think we'll have to go into detail on these first if we were to use/cover them.

Yeah I agree I saw a lot of image datasets. And a lot of structured EHR data too requires atleast a login like MIMICIII. Yeah I'll look into getting SEER/other genomic data! That would be fun. And yeah no worries the methods aren't that hard. What do you think about us using Google Genomics?

**Comment by m3ngineer** [comment]
It looks like the public datasets on GG can be pretty broad (1000 genomes).
I think if we use a purely genomic dataset, then we should have a tailored
question/problem to solve so its focused.

**Comment by ngiangre** [comment]
Agreed. And honestly it's more data we don't have to handle. We can get smaller data from more feasibly-retrievable sources. 

I can work on that. Do you want to delineate some out-of-the-box methods that we'd like to apply/explore? We can reconvene by Monday August 12th. Sound good?

**Comment by m3ngineer** [comment]
Sounds good, thanks nick!

**Comment by ngiangre** [comment]
Thank YOU! For the community!

**Comment by ngiangre** [comment]
Hey @m3ngineer,

The best data for 'omics that's curated is from the cancer genome atlas [TCGA](https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga), and luckily it's easily accessible from the Bioconductor package manager in R. And I show loading of mmutation and clinical datasets, but there's also RNA, methylation, mkicroRNA, and more datasets as easy as a few commands in the notebook! It's in R, but we can always output to a data folder and load in python.

Check out https://github.com/nyhais/hacknights/blob/master/Bioconductor.ipynb

I also added a single cell RNASeq dataset and a mutation dataset from [1000 genomes](https://www.internationalgenome.org/).

Looking forward to your update!

Cheers,
Nick

**Comment by m3ngineer** [comment]
Awesome,

I found a couple of previous walkthroughs where people have worked with a dataset from TGCA:

Classification of tumor samples based on RNA-seq data:
https://medium.com/@alenaharley/tumor-classification-using-gene-expression-data-poking-at-a-problem-using-fast-ai-again-8633c2256c85

Classifying primary origin of tumors based on DNA point mutations:
https://towardsdatascience.com/the-mystery-of-the-origin-cancer-type-classification-using-fast-ai-libray-212eaf8d3f4e

Also, I believe this HCC dataset has some survival data that I'm going to try to find. If we have that, it could be a cool project to look at genetics affecting overall survival. 
https://portal.gdc.cancer.gov/projects/TCGA-LIHC
https://www.cbioportal.org/study/summary?id=lihc_tcga
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6484308/

**Comment by m3ngineer** [comment]
@ngiangre Confirming there's survival data in there, plus a lot of other information (eg, therapy, expression, imaging). There's a number of papers using this dataset that we could use as inspiration, but not sure how time consuming it is to prep this genomic data. How about a call sometime next week to align on an idea to pursue?

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6329279/
https://www.ncbi.nlm.nih.gov/pubmed/31139015
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6528264/
https://www.ncbi.nlm.nih.gov/pubmed/30247807

**Comment by ngiangre** [comment]
@m3ngineer This is great!!

Yes let's brainstorm I think we're now in a good place. I'm free Tuesday and Wednesday nights next week. Do you want to email me a calendar invite for a video call? Anytime after 6pm those days works for me.

**Comment by m3ngineer** [comment]
Geeat, let’s plan for wed!

**Comment by ngiangre** [comment]
See here:

https://github.com/nyhais/hacknights/blob/master/EDA_TCGA_Liver_Cancer_ngiangre.ipynb

This is the ntbk I'll be adding to/working in. Check if those are the cancers we want to investigate. Thanks!

**Comment by ngiangre** [comment]
Thought this might be interesting for us to look at https://www.biorxiv.org/content/10.1101/778472v1.abstract?%3Fcollection=. 

P.S. they are getting back to us a revised blurb by tomorrow.

**Comment by ngiangre** [comment]
@m3ngineer 
Let me know what you think of https://github.com/nyhais/hacknights/blob/master/Hacknights_TCGA_EDA.ipynb

**Comment by m3ngineer** [comment]
@ngiangre awesome! I added some context text to some of the sections. Were you ever able to create a wrapper for using DEseq2 with python?

**Comment by ngiangre** [comment]
@m3ngineer great! I'm currently getting it to the point of building a wrapper but not quite there yet...I think tomorrow I'll atleast have different differential signatures outputted and have an executable, if not have a python subprocess call the executable within the notebook and read in the output. 

The thing is, it's really slow - like it's taking 25 minutes to compute differential genes between males and females...what might be best is just to have different signatures available to load in. What do you think?

**Comment by m3ngineer** [comment]
Is it possible to just put in just a subset of the genes?

**Comment by ngiangre** [comment]
Yeah. Let me see if that speeds things up. Do we want to have that as an option?

 

Cheers,

Nick

**Comment by m3ngineer** [comment]
If it's possible it might be a good option for at least allowing them to
explore different conditions, even if it doesn't include all genes. I think
even looking at 1-2k at a time could be useful.

**Comment by ngiangre** [comment]
Ok I’ll see what I can do by tomorrow. I’ll message you then. 

 

Cheers,

Nick

**Comment by m3ngineer** [comment]
Cool, if it happens it happens, but any output will work!

**Comment by ngiangre** [comment]
For 1000 genes, takes 2 minutes. So you think a random sample option is fine or should I do top variable or most expressed genes?

 

Cheers,

Nick

**Comment by ngiangre** [comment]
pull hacknights and, within hacknights/, execute `Rscript TCGA_differential_expression_analysis.R "" lihc_rnaseq.csv.gz lihc_clinical.csv.gz 100 patient.gender female male`. Let me know if that outputs a file to that same directory.

**Comment by m3ngineer** [comment]
@ngiangre it worked! Took about a min for the 100 genes
